src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/MessageProcessor.java

```diff
@@ -24,15 +24,20 @@
 
     @Scheduled(fixedDelayString = "${queue.read-interval}")
     public void run() {
+        logger.info("Started processing queue messages.");
+
         IMessage msg = null;
+        int processedMessagesCount = 0;
+
         try {
             IMessageReceiver msgReceiver = messageReceiverFactory.create();
             while ((msg = msgReceiver.receive()) != null) {
-
                 envelopeProcessor.onMessageAsync(msg).get();
-
                 msgReceiver.complete(msg.getLockToken());
+                processedMessagesCount++;
             }
+
+            logger.info("Message processing complete. Processed {} messages", processedMessagesCount);
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
```

Comment:
this also needs to be counted.

Reply:
What do you mean?

Reply:
can be easily counted via appinsights taking into account the line below: `logger.error("interrupted", e)`. this will be registered with exceptions and can be filtered smth like `message == "interrupted"`